### ISSUE TITLE
tgui: Unhandled promise rejection is not properly forwarded to window.onerror

### ISSUE THREAD

**Issue by AMARTI15** [tgstation/tgstation#55588]
Was told to send this to an Authorized Nanotransen incident handler.

TGUI. The current application will be terminated. Send the copy of the following stack trace to an authorized Nanotrasen incident handler at https://github.com/tgstation/tgstation. Thank you for your cooperation. 
 [object Event]
   at undefined:undefined
User Agent: Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.2; WOW64; Trident/7.0; .NET4.0C; .NET4.0E; .NET CLR 2.0.50727; .NET CLR 3.0.30729; .NET CLR 3.5.30729; NMTE; Zoom 3.6.0)
State: {"ckey":"aciiimart","window":{"fancy":0,"locked":0}}

Nanotrasen (c) 2525-2559

**Comment by dragomagol** [comment]
I'm confused. Is the entire body of this issue an error message you saw? What happened that caused it? What round was it in? We can't really diagnose these problems without a clearer picture.

**Comment by stylemistake** [comment]
Yep, that's an error, thanks for sending it in.

**Comment by AMARTI15** [comment]
Yeah it's just an error that popped up after a round ended. I could be
mistaken but I think the round ID is in there. BYOND closed after the round
happened and I got that message but the next time I tried to open it up I
had no problems. But yes, the entire message I just copy pasted and it
showed up in the text box after the round end summary popped up I thought I
should follow the instructions and pass it on.
